src/utils/calc.rs

```diff
@@ -95,12 +95,17 @@ impl SlotCalculator {
 
     /// The current slot number.
     pub fn current_slot(&self) -> u64 {
-        self.calculate_slot(chrono::Utc::now().timestamp() as u64)
+        self.calculate_slot(self.current_timestamp())
+    }
+
+    /// The current timestamp in seconds.
+    pub fn current_timestamp(&self) -> u64 {
```

Comment:
doesn't use `self` in the function body, shouldn't be a method. also wouldn't use the `Self` type, so shouldn't be an associated function